appservice/src/tryGetSiteResource.ts

```diff
@@ -28,11 +28,19 @@ export async function tryGetWebAppSlot(client: WebSiteManagementClient, resource
  * 3. { "Code": "NotFound", "Message": "Server farm with name appsvc_linux_centralus not found." }
  */
 async function tryGetSiteResource<T>(callback: () => Promise<T | WebSiteManagementModels.DefaultErrorResponse>): Promise<T | undefined> {
```

Comment:
I don't think `tryGetSiteResource` is the appropriate place to fix this issue. It's specifically meant to workaround an azure-sdk-for-js issue as described in the comment above and it affects a lot of code. I think your fix needs to be more targeted to specific scenarios.

What's the user experience like with your fix?  For example, if I look at [this code](https://github.com/microsoft/vscode-azuretools/blob/main/appservice/src/createAppService/AppServicePlanCreateStep.ts#L31) it appears that `tryGetAppServicePlan` will return `undefined` and we will try to create a new plan - which wouldn't work.

Reply:
I agree the current commit doesn't resolve the issue, when I just tested it. 
However, the modification at `tryGetSiteResource` is required as part of the fix, as `AuthorizationFailed` errors are thrown from the callback() instead of being returned as a DefaultErrorResponse as they are expected to behave.

Reply:
The only case that should return undefined for `tryGetSiteResource` is if the resource doesn't exist. Every other error (`AuthorizationFailed`, `ECONNRESET`, timeout, etc.`) _should_ be thrown as an error. Nothing should be returned as `DefaultErrorResponse`.

Reply:
Ok I see what you mean, that's why only the error "NotFound" returns undefined.